packages/@aws-cdk/aws-s3-deployment/test/bucket-deployment.test.ts

```diff
@@ -1136,16 +1160,7 @@ test('using deployment bucket references the destination bucket by means of the
       Value: { 'Fn::GetAtt': ['DeploymentCustomResource47E8B2E6', 'DestinationBucketArn'] },
     },
     DestinationName: {
-      Value: {
-        'Fn::Select': [0, {
-          'Fn::Split': ['/', {
-            'Fn::Select': [5, {
-              'Fn::Split': [':',
-                { 'Fn::GetAtt': ['DeploymentCustomResource47E8B2E6', 'DestinationBucketArn'] }],
-            }],
-          }],
-        }],
-      },
+      Value: { Ref: 'DestC383B82A' },
```

Comment:
This is a nice side effect. We can now use the correct reference instead of parsing the bucket name from the ARN.